### ISSUE TITLE
Package Control throw an error in ST2 : can't execute git.

### ISSUE THREAD

**Issue by kpym** [wbond/package_control#995]
I have ST2 and ST3, both with Package Control. No problem in ST3. 
But yesterday (I think) Package Control started posing problem in ST2.

> error: Package Control
> 
> Error executing: "C:\Program Files\Git\cmd\git.exe" config --get branch.perso.remote
> 
> VCS-based packages can be ignored with the "ignore_vcs_packages" setting.
> error: Package Control
> 
> Error executing: "C:\Program Files\Git\cmd\git.exe" config --get branch.perso.merge
> 
> VCS-based packages can be ignored with the "ignore_vcs_packages" setting.
> Exception in thread Thread-5:
> Traceback (most recent call last):
>   File ".\threading.py", line 532, in __bootstrap_inner
>   File ".\package_control\commands\install_package_command.py", line 46, in run
>   File ".\package_control\package_installer.py", line 100, in make_package_list
>   File ".\package_control\upgraders\git_upgrader.py", line 103, in incoming
>   File ".\package_control\upgraders\git_upgrader.py", line 64, in get_working_copy_info
> AttributeError: 'bool' object has no attribute 'replace'

And I can't install, upgrade, ... packages.

**Comment by wbond** [comment]
For a temporary workaround, do what the error suggests and set the `ignore_vcs_packages` setting to `true` in your `Package Control.sublime-settings` file.

I will look into this shortly.

**Comment by PeterPablo** [comment]
For me this also happens with ST3 (on Mac, Windows and Linux).

**Comment by tkalfigo** [comment]
Can confirm this happens to me with ST3 as well. 

Could it be it's because packages are synced across ST installations over Dropbox?

**Comment by tkalfigo** [comment]
Can confirm this happening on ST3/Linux as well. Could it be it's because the packaged are synced over Dropbox across ST installations?

**Comment by ummmmm** [comment]
Bump.  Is an issue for me as well.

**Comment by vedmichv** [comment]
Still reproduce on Mac

**Comment by cundi** [comment]
I got this issue on Mac OS 10.11.2, too. Do we have any solution to deal with it?

**Comment by robmarano** [comment]
I experienced this error on ST3 on Mac OS 10.10.5 too. I have yet to find the problem. Any advice/pointers greatly appreciated.

**Comment by philippotto** [comment]
Any updates on this one? Setting ignore_vcs_packages to true, doesn't fix it for me.

**Comment by Shulin** [comment]
Setting that option didn't fix it for me either. This is preventing me from installing packages.

**Comment by ummmmm** [comment]
You guys need to add **"ignore_vcs_packages": true** to the Package Control.sublime-settings file...

Preferences->Package Settings->Package Control->Settings -- User

**Comment by Shulin** [comment]
Did you read our comments? Setting that to true didn't fix the issue. http://puu.sh/mYTpC/447688a1aa.png

**Comment by ummmmm** [comment]
@Shulin I've seen people put it in the preferences file before, so I was only clarifying. I had this issue too, and putting it in the package control setting file fixed it for me.

**Comment by wbond** [comment]
Anyone having this issue with `ignore_vcs_packages` set to `true` care to debug?

**Comment by philippotto** [comment]
I think I was missing a restart after setting ignore_vcs_packages to true. It's working for me now.

**Comment by htmlstream** [comment]
@wbond Yes, works like a charm! Thank you for the solution!
@philippotto +1

**Comment by Shulin** [comment]
Well, it's working for me again. Maybe it was a restart like @philippotto mentioned. But also, for whatever reason, Package Control was liked in my 'ignored_packages' section of my Preferences.sublime-settings file. So I removed it from there. 

There was also a ST3 update today that I got. 

Not sure which of those things fixed it, but I'm able to use Package Control again. :/

**Comment by ayooby** [comment]
i had same issue in ST3/linux and by setting "ignore_vcs_packages": true it OK now

**Comment by eksperimental** [comment]
creating a remote branch fixes the issue

**Comment by osadan** [comment]
happens to me to. 
event after I set up ignore_vcs_packages : true,
event after restart and etc.. 
happens every time I startup the sublime

**Comment by omus** [comment]
I was also running into this problem. The `ignore_vcs_packages: true` wasn't working for me and I was getting a variety of error messages including:
```
Error executing: /usr/bin/git config --get branch.NAME.remote
Error executing: /usr/bin/git config --get branch.NAME.merge
Error executing: /usr/bin/git symbolic-ref -q HEAD
```
The problem for me was occurring because I had a Git repo cloned into my Sublime package directory. In this repo I had a branch called "NAME" which did not have a remote branch. I fixed the issue by checking out a remote branch and having a local branch with the same name:
```
git checkout -b BRANCH origin/BRANCH
```
It would be nice if package control would report the package which caused the error to be raised.

**Comment by wbond** [comment]
This should be fixed by c209f069785af1416892de10c143348a2a9a804a.

The commit 1a9db4a4286530d2ba24045f807c4a67df321137 should improve error messages by providing the working directory in error dialogs.
